rusk-abi/tests/lib.rs

```diff
@@ -230,27 +234,32 @@ fn plonk_proof() {
 
     let label = b"dusk-network";
 
-    let (prover, verifier) = Compiler::compile(&pp, label)
+    let (prover, verifier) = Compiler::compile::<TestCircuit>(&pp, label)
         .expect("Circuit should compile successfully");
 
-    let circuit = TestCircuit::new(1, 2);
+    let a = 1u64;
+    let b = 2u64;
+    let expected_pi = vec![BlsScalar::from(a) + BlsScalar::from(b)];
+    let circuit = TestCircuit::new(a, b);
 
-    let (proof, public_inputs) = prover
+    let (proof, prover_pi) = prover
         .prove(&mut OsRng, &circuit)
         .expect("Proving circuit should succeed");
 
+    // Check public inputs
+    assert_eq!(
+        expected_pi, prover_pi,
+        "Prover generates different pi than expected"
+    );
+
     // Integrity check
     verifier
-        .verify(&proof, &public_inputs)
+        .verify(&proof, &expected_pi)
         .expect("Proof should verify successfully");
     let verifier = verifier.to_bytes();
 
-    let public_inputs: Vec<PublicInput> = public_inputs
-        .into_iter()
-        // FIXME: this should only be From::from, but due to the negative PI
-        //  problem we invert them here
```

Comment:
Do you have an issue for this? if so, put a `See also #<issue_number>` in the commit to reference it

Reply:
will add link to the commit but adding it here as well for future reference:
https://github.com/dusk-network/plonk/issues/717